libraries/app/database_api.cpp

```diff
@@ -1186,62 +1238,24 @@ market_ticker database_api_impl::get_ticker( const string& base, const string& q
    FC_ASSERT( assets[1], "Invalid quote asset symbol: ${s}", ("s",quote) );
 
    const fc::time_point_sec now = _db.head_block_time();
-
-   market_ticker result;
-   result.time = now;
-   result.base = base;
-   result.quote = quote;
-   result.latest = "0";
-   result.lowest_ask = "0";
-   result.highest_bid = "0";
-   result.percent_change = "0";
+   market_ticker mt;
 
    auto base_id = assets[0]->id;
    auto quote_id = assets[1]->id;
    if( base_id > quote_id ) std::swap( base_id, quote_id );
 
-   fc::uint128 base_volume;
-   fc::uint128 quote_volume;
-
    const auto& ticker_idx = _db.get_index_type<graphene::market_history::market_ticker_index>().indices().get<by_market>();
    auto itr = ticker_idx.find( std::make_tuple( base_id, quote_id ) );
    if( itr != ticker_idx.end() )
    {
-      price latest_price = asset( itr->latest_base, itr->base ) / asset( itr->latest_quote, itr->quote );
-      if( itr->base != assets[0]->id )
-         latest_price = ~latest_price;
-      result.latest = price_to_string( latest_price, *assets[0], *assets[1] );
-      if( itr->last_day_base != 0 && itr->last_day_quote != 0 // has trade data before 24 hours
-            && ( itr->last_day_base != itr->latest_base || itr->last_day_quote != itr->latest_quote ) ) // price changed
-      {
-         price last_day_price = asset( itr->last_day_base, itr->base ) / asset( itr->last_day_quote, itr->quote );
-         if( itr->base != assets[0]->id )
-            last_day_price = ~last_day_price;
-         result.percent_change = price_diff_percent_string( last_day_price, latest_price );
-      }
-      if( assets[0]->id == itr->base )
-      {
-         base_volume = itr->base_volume;
-         quote_volume = itr->quote_volume;
-      }
-      else
-      {
-         base_volume = itr->quote_volume;
-         quote_volume = itr->base_volume;
+      optional<order_book> orders;
+      if (!skip_order_book) {
+         orders = get_order_book(assets[0]->symbol, assets[1]->symbol, 1);
       }
+      mt = market_ticker(*itr, now, *assets[0], *assets[1], orders);
+      return mt;
    }
-
-   result.base_volume = uint128_amount_to_string( base_volume, assets[0]->precision );
-   result.quote_volume = uint128_amount_to_string( quote_volume, assets[1]->precision );
-
-   if( !skip_order_book )
-   {
-      const auto orders = get_order_book( base, quote, 1 );
-      if( !orders.asks.empty() ) result.lowest_ask = orders.asks[0].price;
-      if( !orders.bids.empty() ) result.highest_bid = orders.bids[0].price;
-   }
-
-   return result;
+   return mt;
```

Comment:
Need to include order data if skip_order_book is false, even when no ticker data.